### PR TITLE
Create directory when adding a book section from book tree view

### LINKED ISSUE #18741: When creating a Jupyter Section the section folder is not created
The section folder is not created and files are added to the root folder instead.

### PR DESCRIPTION
**Pull request by barbaravaldez**
Pass in the section name in the file path and create the section directory.
This PR fixes #18741

### CODE REVIEW

**Review by corivera — APPROVED** [review]

File: extensions/notebook/src/book/bookTocManager.ts [hunk 1]
```diff
@@ -470,8 +469,10 @@ export class BookTocManager implements IBookTocManager {
 		};
 
 		if (isSection) {
+			await fs.mkdir(path.dirname(pathDetails.filePath));
 			fileEntryInToc.sections = [];
 		}
+		await fs.writeFile(pathDetails.filePath, '');
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/book/bookTocManager.ts:475]: What is the file that's being written when we're adding a section? Wouldn't a section just be a folder?

Comment (barbaravaldez) [reply extensions/notebook/src/book/bookTocManager.ts:475]: Currently, when we load a book we need a file or url on every toc entry and if we want to make it a section we add the `sections` keyword in the toc.

Comment (Charles-Gagnon) [reply extensions/notebook/src/book/bookTocManager.ts:475]: So if I have a section `/path/to/sectionfoo`

Then we'll end up with both a folder named `sectionfoo` and inside that a file also named `sectionfoo`?

Comment (barbaravaldez) [reply extensions/notebook/src/book/bookTocManager.ts:475]: We let the user enter the file name in the dialog, so it should be `/path/to/sectionfoo/file_name`

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**